build/build-report.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs');
+
+async function buildStandaloneReport() {
+  const REPORT_JAVASCRIPT = [
+    fs.readFileSync(__dirname + '/../report/renderer/util.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/dom.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/details-renderer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/crc-details-renderer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/snippet-renderer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/element-screenshot-renderer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../lighthouse-core/lib/file-namer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/logger.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/report-ui-features.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/category-renderer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/performance-category-renderer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/pwa-category-renderer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/report-renderer.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/i18n.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/renderer/text-encoding.js', 'utf8'),
+    fs.readFileSync(__dirname + '/../report/clients/standalone.js', 'utf8'),
+  ].join(';\n');
+  fs.mkdirSync(__dirname + '/../report/generated', {recursive: true});
```

Comment:
why not in `dist/`?

Reply:
Because we still need to publish it

Reply:
OK. Is there a doc for this?

Reply:
[doc](https://docs.google.com/document/d/1E0dwjbItFFXLJk3TjFr1dHbXLqkPuTwVrhwvbbigp4I/edit?ts=60da95c8&resourcekey=0-oDI_CvqEo3QjX7r0ca3hMA#heading=h.ldv5z9maudqb)

would you prefer the built file to live in `dist/report/` (and be published from there)? no preference here.

Reply:
I also don't have a strong preference. On the one hand, `dist/` isn't for npm published artifacts today, so mixing in published artifacts isn't great. OTOH, a new folder for bundled assets might be confusing too, although `generated` folder name minimizes that IMO.

Reply:
I'll preface with if it ends up in `report/generated/`, I'll definitely live :)

For me, the argument in favor of living under `dist/` is that I could drop into the code base after being away for a while, notice that the report actually has a build step now, but still have a good idea of where the built artifact was going to end up without having to look at any code, because almost everything that we build like this ends up in `dist/`. I think there's also an argument to be made for a top-level directory of built things being referred to by checked-in code vs a subdirectory that will get filled in after building in terms of code clarity, but maybe that's the same thing phrased a different way.

It does make `.npmignore` more awkward, but that's really just a slight awkwardness for us that I think we'll quickly forget. If I were to drop into a package in `node_modules/` and it had a solitary top-level `dist/report/` directory in it, I wouldn't think that was weird at all.

Reply:
Moved to `dist/report`